### PR TITLE
Switch to thin LTO on windows

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,5 +1,9 @@
+@echo on
 set "PYO3_PYTHON=%PYTHON%"
 
+sed -i "/lto = ""fat""/lto = ""thin""/g" Cargo.toml
+if errorlevel 1 exit 1
+
 set "CMAKE_GENERATOR=NMake Makefiles"
 set "RUSTFLAGS=-C codegen-units=4"
```
Comment (ritchie46) [inline recipe/bld.bat:8]: I think using a single thread also consumes less memory. 

The output binaries compiled with a single thread are also faster.

Comment (xhochy) [reply recipe/bld.bat:8]: i.e. `codegen-units=1`?

Comment (ritchie46) [reply recipe/bld.bat]: Yeap!

**Review by timkpaine — APPROVED** [review]
Lgtm, presumeably you didn't need to retry any of the builds so hopefully this resolves it

**Pull request merged**